turbinia/workers/analysis/yara.py

```diff
@@ -43,6 +47,26 @@ class YaraAnalysisTask(TurbiniaTask):
       'minscore': None
   }
 
+  def update_rules(self):
+    """Update the Yara rules.
+
+    Args: None
+        
+    Returns: None
+    """
+    rules = {
+        'https://github.com/Neo23x0/signature-base.git': '/opt/signature-base'
+    }
+    log.debug('Updating Yara rules')
+    for repo, path in rules.items():
+      try:
+        repository = git.Repo(path)
+        origin = repository.remotes.origin
+        origin.pull(depth=1)
+        log.debug('Successfully updated rules from %s in %s', repo, path)
+      except Exception as e:
```

Comment:
I have added tests and checked for the specific InvalidGitRepositoryError error. I and still catching all other errors as we want to indeed fail open. As git is complicated it can return many different errors -> https://gitpython.readthedocs.io/en/stable/reference.html#module-git.exc